frontend/talentsearch/src/js/components/workLocationPreferenceForm/WorkLocationPreferenceForm.test.tsx

```diff
@@ -0,0 +1,137 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from "react";
+import "@testing-library/jest-dom";
+import { IntlProvider, MessageFormatElement } from "react-intl";
+import { fakeUsers } from "@common/fakeData";
+import { WorkLocationPreferenceQuery, WorkRegion } from "../../api/generated";
+import { render, screen, fireEvent, act, waitFor } from "../../tests/testUtils";
+import {
+  WorkLocationPreferenceForm,
+  WorkLocationPreferenceFormProps,
+} from "./WorkLocationPreferenceForm";
+
+const renderWithReactIntl = (
+  component: React.ReactNode,
+  locale?: "en" | "fr",
+  messages?: Record<string, string> | Record<string, MessageFormatElement[]>,
+) => {
+  return act(() => {
+    render(
+      <IntlProvider locale={locale || "en"} messages={messages}>
+        {component}
+      </IntlProvider>,
+    );
+  });
+};
+
+const mockUser = fakeUsers()[0];
+
+const mockInitialData: WorkLocationPreferenceQuery | undefined = {
+  __typename: "Query",
+  me: {
+    __typename: "User",
+    id: "thanka11",
+    locationPreferences: [WorkRegion.Atlantic],
+    locationExemptions: "dagu",
+  },
+};
+const mockInitialEmptyData: WorkLocationPreferenceQuery | undefined = {
+  __typename: "Query",
+  me: {
+    __typename: "User",
+    id: "thanka11",
+    locationPreferences: undefined,
+    locationExemptions: "",
+  },
+};
+
+const renderWorkLocationPreferenceForm = ({
+  initialData,
+  handleWorkLocationPreference,
+}: WorkLocationPreferenceFormProps) => (
+  <>
+    {renderWithReactIntl(
+      <WorkLocationPreferenceForm
+        initialData={initialData}
+        handleWorkLocationPreference={handleWorkLocationPreference}
+      />,
+    )}
+  </>
+);
```

Comment:
I noticed there's a way to make the test file just a bit more concise. If you look at the testUtils file, you'll see that the `render` function it exports actually already wraps whatever you pass in with an IntlProvider component. Instead of defining `renderWithReactIntl` and `renderWorkLocationPreferenceForm`, you could simply call
```
render(
    WorkLocationPreferenceForm
      initialData={initialData}
      handleWorkLocationPreference={handleWorkLocationPreference}
    />
 )
 ````

Reply:
See SkillChecklist.test.tsx and SkillResrults.test.tsx for examples

Reply:
@esizer just making sure you see the tip about testUtils as well!

Reply:
Ok thank you! I just follow some of the existing testing including Eric's. I will take a look.

Reply:
> @esizer just making sure you see the tip about testUtils as well!

Nice! Thanks for the tip.